test/jdk/java/awt/Scrollbar/ScrollbarKeyControlTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4943277
+  @requires (os.family == "linux")
+  @summary XAWT: Scrollbar can't be controlled by keyboard
+  @key headful
+*/
+
+import java.awt.AWTException;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.Scrollbar;
+import java.awt.Toolkit;
+
+import java.awt.event.AdjustmentListener;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+
+public class ScrollbarKeyControlTest implements AdjustmentListener, KeyListener {
+    Scrollbar scrollbarV;
+    Scrollbar scrollbarH;
+    volatile int changesTotal = 0;
+    Robot robot;
+    Object LOCK = new Object();
+    Frame frame;
+
+    public static void main(String[] args) throws Exception {
+
+        final boolean isXAWT = Toolkit.getDefaultToolkit().getClass().getName().
+                equals("sun.awt.X11.XToolkit");
+
+        if (!isXAWT) {
+            System.out.println("This test is for XAWT only.");
+            return;
+        }
```

Comment:
This is largely pointless since you have requires os==linux.
Perhaps its from when we had MToolkit.
If you keep it, then I'd make it just check the "os.name" and not the toolkit

Reply:
Does this test work on mac or windows? if not any reason why?